### PR TITLE
Remote guardian sign

### PR DESCRIPTION
**Pull request by jyouyj**
Adjust the submit_work RPC to support remote guardian sign.

### CODE REVIEW

File: quarkchain/cluster/miner.py [hunk 1]
```diff
@@ -279,7 +281,7 @@ def _mine_new_block_async(self):
             copy.deepcopy(self.current_work),
         )
 
-    async def submit_work(self, header_hash: bytes, nonce: int, mixhash: bytes) -> bool:
+    async def submit_work(self, header_hash: bytes, nonce: int, mixhash: bytes, signature: bytes = bytes(65),) -> bool:
```
Comment (qizhou) [inline quarkchain/cluster/miner.py]: bytes(65),) => bytes(65))  (Remove unused ,)

File: quarkchain/cluster/tests/test_miner.py [hunk 2]
```diff
@@ -213,6 +213,7 @@ def test_submit_work(self):
         loop = asyncio.get_event_loop()
         loop.run_until_complete(go())
 
+
```
Comment (qizhou) [inline quarkchain/cluster/tests/test_miner.py]: Remove this line

File: quarkchain/cluster/tests/test_jsonrpc.py [hunk 3]
```diff
@@ -857,6 +857,7 @@ def test_getWork_and_submitWork(self):
                 clusters[0].get_shard_state(1 | 0).get_tip().header.height, 1
             )
 
+
```
Comment (qizhou) [inline quarkchain/cluster/tests/test_jsonrpc.py]: Remove

File: quarkchain/cluster/jsonrpc.py [hunk 4]
```diff
@@ -865,15 +871,16 @@ def get_data_default(key, decoder, default=None):
     @decode_arg("header_hash", hash_decoder)
     @decode_arg("nonce", quantity_decoder)
     @decode_arg("mixhash", hash_decoder)
-    async def submitWork(self, full_shard_key, header_hash, nonce, mixhash):
+    @decode_arg("signature", signature_decoder)
+    async def submitWork(self, full_shard_key, header_hash, nonce, mixhash, signature="0x" + bytes(65).hex()):
```
Comment (qizhou) [inline quarkchain/cluster/jsonrpc.py]: signature=bytes(65)?

File: quarkchain/cluster/miner.py [hunk 5]
```diff
@@ -289,11 +291,18 @@ def _mine_new_block_async(self):
         block = copy.deepcopy(self.work_map[header_hash])
         header = block.header
         header.nonce, header.mixhash = nonce, mixhash
-
         # sign as a guardian
         if self.guardian_private_key and isinstance(block, RootBlock):
             header.sign_with_private_key(self.guardian_private_key)
 
+        # remote sign as a guardian
+        if isinstance(block, RootBlock) and header.signature == bytes(65) and signature != bytes(65):
```
Comment (ninjaahhh) [inline quarkchain/cluster/miner.py]: 1. use [precommit + black](https://github.com/QuarkChain/pyquarkchain#development-flow) for code formatting
2. how about following? seems we don't need guardian pub key

```python
if isinstance(block, RootBlock):
  header.signature = signature
```

**Comment by qcdll** [comment]
suggest using black to format the code before putting up for review

File: quarkchain/cluster/jsonrpc.py (hunk 4 above)
Comment (jyouyj) [reply quarkchain/cluster/jsonrpc.py]: I think it is a string. After the @decode_arg("signature", signature_decoder), it is converted to bytes(65).

**Comment by jyouyj** [comment]
I close this pull request since it is the same as #505. Please refer to #505 for more details.

**Pull request closed without merging**